### PR TITLE
Webpack by value

### PR DESCRIPTION
**Pull request by 12944qwerty**
Get the module based on value string/regex.

### CODE REVIEW

File: src/renderer/modules/webpack/index.ts [hunk 1]
```diff
@@ -380,6 +380,25 @@ export namespace filters {
       return typeof match === "string" ? source.includes(match) : match.test(source);
     };
   };
+
+  /**
+   * Get a module that has the given value on one of its exports
+   * @param match The string to check the value against
+   *
+   * @remarks
+   * Great for getting a module for a specific classname
```
Comment (asportnoy) [inline src/renderer/modules/webpack/index.ts:389]: Can you put an example here?

File: src/renderer/modules/webpack/index.ts [hunk 2]
```diff
@@ -380,6 +380,25 @@ export namespace filters {
...
+   */
+  export const byValue = (match: string | RegExp) => {
+    return (m: RawModule) => {
+      if (m.exports) {
+        return typeof match === "string"
+          ? Object.values(m.exports).includes(match)
+          : Object.values(m.exports).some((val) => typeof val === "string" && match.test(val));
+      } else {
+        return false;
+      }
```
Comment (asportnoy) [inline src/renderer/modules/webpack/index.ts:399]: Better to handle the edge cases first
```suggestion
      if (!m.exports) {
        return false
      }

      return typeof match === "string"
        ? Object.values(m.exports).includes(match)
        : Object.values(m.exports).some((val) => typeof val === "string" && match.test(val));
```

**Review by asportnoy — CHANGES_REQUESTED** [review]

File: src/renderer/modules/webpack/index.ts [hunk 3]
```diff
@@ -380,6 +380,29 @@ export namespace filters {
...
+   *
+   * @remarks
+   * Great for getting a module for a specific classname
+   */
+  export const byValue = (match: string | RegExp) => {
+    return (m: RawModule) => {
+      if (!m.exports) {
+        return false;
+      }
+      return typeof match === "string"
```
Comment (asportnoy) [inline src/renderer/modules/webpack/index.ts:401]: module exports is not guaranteed to be an object, so you should check for that here

**Review by asportnoy — CHANGES_REQUESTED** [review]

File: src/renderer/modules/webpack/index.ts [hunk 4]
```diff
@@ -395,7 +395,7 @@ export namespace filters {
    */
   export const byValue = (match: string | RegExp) => {
     return (m: RawModule) => {
-      if (!m.exports) {
+      if (!m.exports && typeof m.exports !== "object") {
```
Comment (asportnoy) [inline src/renderer/modules/webpack/index.ts:398]:
```suggestion
      if (!m.exports || typeof m.exports !== "object") {
```

**Review by asportnoy — CHANGES_REQUESTED** [review]

**Review by asportnoy — APPROVED** [review]
